src/Microsoft.Sbom.Api/Workflows/SBOMParserBasedValidationWorkflow.cs

```diff
@@ -278,4 +286,37 @@ private void LogResultsSummary(ValidationResult validationResultOutput, IEnumera
 
         Console.WriteLine($"Unknown file failures . . . . . . . . . . . . .  {validFailures.Count(v => v.ErrorType == ErrorType.Other)}");
     }
+
+    private void ThrowOnInvalidInputFiles(List<FileValidationResult> fileValidationFailures)
+    {
+        var invalidInputFiles = fileValidationFailures.Where(f => f.ErrorType == ErrorType.InvalidInputFile).ToList();
+        if (invalidInputFiles.Count != 0)
+        {
+            throw new InvalidDataException($"Your manifest file is malformed. {invalidInputFiles.First().Path}");
+        }
+    }
+
+    /// <summary>
+    /// Set compliance standard for SPDX 3.0 parsers and above.
+    /// </summary>
+    /// <param name="spdxVersion"></param>
+    private void SetComplianceStandard(string spdxVersion, ISbomParser sbomParser)
+    {
+        // Note that spdxVersion is already validated to be a valid double in ConfigValidator.
+        var spdxVersionAsDouble = Convert.ToDouble(spdxVersion);
+
+        if (!string.IsNullOrEmpty(configuration.ComplianceStandard?.Value) && spdxVersionAsDouble >= 3.0)
+        {
+            var complianceStandard = configuration.ComplianceStandard.Value;
+            try
+            {
+                (sbomParser as SPDX30Parser).RequiredComplianceStandard = complianceStandard;
```

Comment:
This feels off. We're downcasting (never a good thing), and if we're validating a 2.2 SPDX file, we throw a `NullReferenceException`, record an error, then keep on running. Would it make sense to add `ISbomParser.SetComplianceStandard` and let the specific parser version decide what standards it can consume? If an unsupported standard is specified, there may be an argument to fail the task because the input is broken.

Reply:
I added `ISbomParser.SetComplianceStandard` and adjusted the implementations of the parsers so that should make this a bit cleaner.